File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    for i in range(1, int(yellow**(0.5)+1)):
        if yellow%i==0:
            yellow_w = i
            yellow_h = yellow//i
            if (brown+yellow)==(yellow_h+2)*(yellow_w+2):
                return [yellow_h+2, yellow_w+2]
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.java
/**
w:4, h:3   -> 테두리 10 (4*2 + 3*2 -4),
w:3, h:3   -> 테두리 8  (3*2 + 3*2 -4)
w:8, h:6   -> 테두리 24 (w*2 + h*2 -4)

**/
import java.util.*;

class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {0, 0};
        int size = brown+yellow;
        
        for(int h=1; h<=(int)Math.sqrt(size); h++){
            if(size%h==0){
                int w = size/h;
                if(brown== (w+h)*2-4){
                    answer[0] = w;
                    answer[1] = h;
                    break;
                }
            }
        }
        return answer;
    }
}